reflex/.templates/apps/basetemplate/code/basetemplate.py

```diff
@@ -0,0 +1,72 @@
+"""Welcome to Reflex! This file outlines the steps to create a basic app."""
+import reflex as rx
+
+from .sidebar import sidebar
+from .styles import *
+
+
+def content():
+    return rx.box(
+        rx.vstack(
+            rx.heading(
+                "Dashboard",
+                size="3em",
+            ),
+            rx.text(
+                "Welcome to Reflex!",
+            ),
+            rx.text(
+                "You can use this template to get started with Reflex.",
+            ),
+            width="100%",
+            align_items="flex-start",
+            height="90%",
+            box_shadow="0px 0px 0px 1px rgba(84, 82, 95, 0.14)",
+            border_radius=border_radius,
+            padding="1em",
+        ),
+        height="100vh",
+        width="100%",
+        padding_top="5em",
+        padding_x="2em",
+    )
+
+
+def menu_button():
+    return rx.box(
+        rx.menu(
+            rx.menu_button(
+                rx.icon(
+                    tag="hamburger",
+                    size="4em",
+                    color=text_color,
+                ),
+            ),
+            rx.menu_list(
+                rx.menu_item("Home"),
+                rx.menu_divider(),
+                rx.menu_item("About"),
+                rx.menu_item("Contact"),
+            ),
+        ),
+        position="fixed",
+        right="1.5em",
+        top="1.5em",
+        z_index="500",
+    )
+
+
+def index() -> rx.Component:
+    return rx.hstack(
+        sidebar(),
+        content(),
+        rx.spacer(),
+        menu_button(),
+        align_items="flex-start",
+    )
```

Comment:
we should expose this as a decorator

```suggestion
def template(main_content: Callable[[], rx.Component]) -> rx.Component:
    return rx.hstack(
        sidebar(),
        main_content(),
        rx.spacer(),
        menu_button(),
        align_items="flex-start",
    )
```

that way adding new pages can be much simpler

```python
@rx.page("/settings")
@template
def settings():
    return rx.vstack(
        rx.hstack(rx.text("Volume"), rx.slider(width="15em")),
        rx.hstack(rx.text("Distortion"), rx.switch()),
        rx.hstack(rx.text("Reverb"), rx.switch()),
        width="100%",
    )
```

... but even that is weird, because i don't have the style props from `content` that pad it from the top of the page and add the nice box shadow. It's not immediately obvious how to extend the template without doing a lot of monkeying and copy-paste.

I think the core template objects should be in a separate module, and just having content in this `basetemplate.py`. It should be dead simple to add new pages and link them into the sidebar. We should be exposing these interfaces as decorators where possible.

The template should avoid mixing content and presentation logic, we have some stuff in `styles.py`, but there are other essential styles inlined, and it's not clear if i should be copying those when creating new pages, etc.